builder/builder-next/worker/containerdworker.go

```diff
@@ -17,11 +20,21 @@ type ContainerdWorker struct {
 }
 
 // NewContainerdWorker instantiates a local worker.
-func NewContainerdWorker(ctx context.Context, wo base.WorkerOpt, callbacks exporter.BuildkitCallbacks) (*ContainerdWorker, error) {
+func NewContainerdWorker(ctx context.Context, wo base.WorkerOpt, callbacks exporter.BuildkitCallbacks, rt nethttp.RoundTripper) (*ContainerdWorker, error) {
 	bw, err := base.NewWorker(ctx, wo)
 	if err != nil {
 		return nil, err
 	}
+	hs, err := http.NewSource(http.Opt{
+		CacheAccessor: bw.CacheManager(),
+		Transport:     rt,
+	})
+	if err == nil {
```

Comment:
Note: I'm not returning an error here the same way the graph driver worker doesn't return an error

* https://github.com/moby/moby/blob/f5af46d4d5db591da5ce330c9936baecd1df5d9c/builder/builder-next/worker/worker.go#L121-L129

Reply:
Perhaps that's worth a comment in the code for future reference?

Reply:
Looks like that "don't fail  / don't return error" was added in https://github.com/moby/moby/commit/ca3e3fcf7e66554d20acd990a506d6ee5e368ca1 


- https://github.com/moby/moby/pull/37280
- to fix https://github.com/moby/moby/issues/37278

I guess it somewhat makes sense  to not  prevent the daemon from  starting, but haven't tried what it looks like when trying to build something that needs git and/or if BuildKit has some entitlement to indicate whether a builder can support git (or not).

Reply:
So, it looks like that fix was intended for `git`, but now wondering if  it was much  too broad in scope as it also did the same for  other type of sources which (I  _THINK_) don't depend on any external dependency to be installed? 🤔

Reply:
Thanks for digging in! Ok yeah that fix added the warning for the http source for some reason too... I can change c8d and GD cases and return an error for the http source

Reply:
I'd be fine with keeping  them the same for now, then (in a follow-up) fix it for both. Or at least; I haven't looked yet if there's possible reasons for this to fail (and if  we're considering including this in  27.x, perhaps it's safer to   "fail soft" for now, and to look more in depth if we can safely fail loud 😅 


Curious though; should those other sources also be registered? (`git.NewSource`, `local.NewSource`) ?

Reply:
The "funny" part is that `http.NewSource` never returns an error :D

Reply:
> Curious though; should those other sources also be registered? (git.NewSource, local.NewSource) ?

All of these are already added by the call to `base.NewWorker` a couple lines above. We only need to change the http one for the roundtripper to do its thing

Reply:
> The "funny" part is that http.NewSource never returns an error :D

LOL, yeah, in that case; hard  failure  would be good for sure! And possibly we should change that signature  to  not even _pretend_ there may be an error.

> All of these are already added by the call to base.NewWorker a couple lines above. We only need to change the http one for the roundtripper to do its thing

Ah, thanks!